hebsafeharbor/identifier/signals/hebrew_city_recognizer.py

```diff
@@ -57,27 +59,138 @@ def analyze(
         if len(terms_offsets) == 0:
             return results
 
-        # iterate over the spaCy tokens, and find all the position indices for endorsing entities
-        geo_entities_position_set = set()
-        for entity in nlp_artifacts.entities:
-            if entity.label_ in self.endorsing_entities:
-                for char in range(entity.start_char, entity.end_char):
-                    geo_entities_position_set.add(char)
-
         # iterate over potential city offsets and endorse the results using overlap with geo-entities or support words
         for term_start, term_len in terms_offsets:
-            overlap_ratio = len(
-                set(range(term_start, term_start + term_len)).intersection(geo_entities_position_set)) / term_len
-            adjusted_score = self.DEFAULT_CONFIDENCE_LEVEL + (overlap_ratio * self.LOCATION_OVERLAP_FACTOR)
             result = RecognizerResult(
                 entity_type="CITY",
                 start=term_start,
                 end=term_start + term_len,
-                score=adjusted_score,
-                analysis_explanation=AnalysisExplanation(self.name, adjusted_score),
+                score=self.DEFAULT_CONFIDENCE_LEVEL,
+                analysis_explanation=AnalysisExplanation(self.name, self.DEFAULT_CONFIDENCE_LEVEL),
                 recognition_metadata={RecognizerResult.RECOGNIZER_NAME_KEY: self.name},
             )
 
             results.append(result)
 
+        results = self.enhance_using_other_entities(nlp_artifacts, results)
+
+        return results
+
+    def enhance_using_context(
+        self,
+        text: str,
+        raw_recognizer_results: List[RecognizerResult],
+        other_raw_recognizer_results: List[RecognizerResult],
+        nlp_artifacts: NlpArtifacts,
+        context: Optional[List[str]] = None,
+    ) -> List[RecognizerResult]:
+        """Overrides generic context enhancer with recognizer-specific.
+        Enhance confidence score using context of the entity.
+
+        in case a result score is boosted, derived class need to update
```

Comment:
Consider updating the docstring, this refers to the parent function. Also, worth mentioning why we're doing a recognizer-level context enhancement in this case